web/src/main/java/com/graphhopper/http/GraphHopperServlet.java

```diff
@@ -144,7 +145,7 @@ public void doGet(HttpServletRequest httpReq, HttpServletResponse httpRes) throw
                         put(WAY_POINT_MAX_DISTANCE, minPathPrecision);
 
                 ghRsp = graphHopper.route(request);
-            } catch (IllegalArgumentException ex) {
+            } catch (RuntimeException ex) {
```

Comment:
As explained: without further work this change is not possible as for all other exceptions we have to throw http error == 500

Reply:
Thanks, I added additional handling here. Would this be enough or should it work differently?

Reply:
Would you ensure this via an integration test (illegal arg => 400, trigger exception => 500). Triggering an exception is probably only possible via injecting a 'broken' GraphHopper class...

Reply:
Yes makes sense. I looked a bit more into this and this might be a bit complicated. Injecting a GraphHopper class is a surprising overhead. I had to do a couple of rather dirty things in the test class, but I could not find a better way to do this. I did not want to change anything in the production code. The issue is that we cannot extend the method `GraphHopperModule#createGraphHopper`, because of the `@Provides` annotation. We could create a wrapper method to make this possible, but I am not sure if we should do this just for a test?

Reply:
You should be able to use

install(Modules.override(new GraphHopperModule(cmdArgs)).with(new TestModule(cmdArgs)));

and then put another createGraphHopper in the TestModule

Reply:
> install(Modules.override(new GraphHopperModule(cmdArgs)).with(new TestModule(cmdArgs)));

I had a look at this, but this would end up in the same amount of code duplication, wouldn't it? Or maybe I didn't understand this correctly?

Reply:
No, you just need to implement createGraphHopper not all the other methods as they are provided from the GraphHopperModule

Reply:
Yes, but I cannot override the createGraphHopper method because of the @Provide annotation or do I miss something?

Reply:
You cannot but because of the Modules.override the provider (if existent) of the test modules are called and not the ones in GraphHopperModule. So use a provide annotation and the GraphHopper return type.